todo-mvc/src/widgets/createTodoItem.ts

```diff
@@ -17,82 +17,125 @@ interface TodoItemState extends WidgetState, StatefulChildrenState {
 
-	checkbox.setState({
-		checked: todoItem.state.completed
+	checkbox.widget.setState({
+		checked: this.state.completed
 	});
 }
 
 const createTodoItem: TodoItemFactory = createWidget
-	.mixin(createParentMixin)
 	.mixin(createRenderableChildrenMixin)
-	.mixin(createStatefulChildrenMixin)
 	.mixin({
-		mixin: createParentMixin,
-		initialize(instance) {
-			const checkbox = createCheckboxInput({
-				state: {
-					id: 'checkbox',
-					classes: ['toggle']
-				},
-				listeners: {
-					change: () => { todoToggleComplete.do(instance.state); }
-				}
-			});
-
-			const button = createButton({
-				state: {
-					id: 'button',
-					classes: ['destroy']
-				},
-				listeners: {
-					click: () => { todoRemove.do(instance.state); }
-				}
-			});
-
-			const label = createWidget({
-				state: {
-					id: 'label'
-				},
-				listeners: {
-					dblclick: () => { todoEdit.do(instance.state); }
-				},
-				tagName: 'label'
-			});
-
-			const editInput = createFocusableTextInput({
-				state: {
-					id: 'editInput',
-					classes: ['edit']
-				},
-				listeners: {
-					blur: (event) => { todoSave.do({state: instance.state, event}); },
-					keyup: (event) => { todoEditInput.do({state: instance.state, event}); }
-				}
-			});
-
-			instance.append([checkbox, button, label, editInput]);
-			instance.on('statechange', manageChildren);
+		mixin: createStatefulChildrenMixin,
+		initialize(instance, options) {
+			instance
+				.createChildren({
+					checkbox: {
+						factory: createCheckboxInput,
+						options: {
+							state: {
+								classes: [ 'toggle' ]
+							},
+							listeners: {
+								change: () => { todoToggleComplete.do(instance.state); }
+							}
+						}
+					},
+					button: {
+						factory: createButton,
+						options: {
+							state: {
+								classes: [ 'destroy' ]
+							},
+							listeners: {
+								click: () => { todoRemove.do(instance.state); }
+							}
+						}
+					},
+					label: {
+						factory: createWidget,
+						options: {
+							listeners: {
+								dblclick: () => { todoEdit.do(instance.state); }
+							},
+							tagName: 'label'
+						}
+					},
+					editInput: {
+						factory: createFocusableTextInput,
+						options: {
+							state: {
+								classes: [ 'edit' ]
+							},
+							listeners: {
+								blur: (event: Event) => { todoSave.do({state: instance.state, event}); },
+								keyup: (event: Event) => { todoEditInput.do({state: instance.state, event}); }
+							}
+						}
+					}
+				})
+				.then((children: TodoItemChildren) => {
+					/* TODO: We are only using the label.widget but we are storing label: { id, widget }, is
+					 * that necessary? */
+					childrenMap.set(instance, children);
+					instance.on('statechange', manageChildren);
+
+					/* We have missed the widgets initial state change because we created the widgets async
+					 * so we should emit a statechange event just to ensure we get the logic we expect
+					 * to be handled */
+					instance.emit({
```

Comment:
Do we think it is a challenge though, that at the moment, we don't have back pressure in `Evented` and it largely aligns to DOM events.  It also always resolves sync, like DOM events.

The challenge too is that it is only Streams that offers back pressure at the moment.  Observables, which would/could be a more logical way to transform `Evented`, but then we don't have that as part of the standard.  Also, if we indiscriminately store every event just in case a listeners comes along later, we would run into a memory issue.

Anyways, I opened dojo/compose#57 to discuss.  Question is, does this stop of from accepting this PR and the other other associated PRs?